js/control-shift-service.js

```diff
@@ -0,0 +1,104 @@
+var urlSeparator = ':';
+var categorySeparator = '|';
+var characterLimit = 200;
+
+var compiledTemplate = Handlebars.compile(getControlShiftTemplate());
+
+var firstPageRequests = [];
+var allPageRequests = [];
+
+var layerGroups = {};
+
+function fetchControlShiftData(categoriesString, callback) {
+  if (!categoriesString) {
+    callback({});
+    return;
+  }
+
+  var categories = categoriesString.split(categorySeparator);
+  
+  var categoryPageNumberMap = getPagesForEachCategory(categories);
+  
+  $.when.apply($, firstPageRequests).done(function() {
+    categories.forEach(function(categoryInfo) {
+      createCampaignMarkersForCategory(categoryInfo, categoryPageNumberMap);
+    });
+
+    renderMarkers(callback);
+  });
+}
+
+function getPagesForEachCategory(categories) {
+  var categoryPageNumberMap = {};
+  
+  categories.forEach(function(categoryInfo) {
+    var domain = categoryInfo.split(urlSeparator)[0];
+    var category = categoryInfo.split(urlSeparator)[1];
+
+    firstPageRequests.push($.ajax({
+      url: getControlShiftUrl(domain, category, 1),
+      dataType: 'jsonp',
+      success: function(data) {
+        categoryPageNumberMap[categoryInfo] = data.total_pages;
+      }
+    }));
+  });
+
+  return categoryPageNumberMap;
+}
+
+function createCampaignMarkersForCategory(categoryInfo, categoryPageNumberMap) {
+  var domain = categoryInfo.split(urlSeparator)[0];
+  var category = categoryInfo.split(urlSeparator)[1];
+  var totalPages = categoryPageNumberMap[categoryInfo];
+
+  for (var i = 1; i <= totalPages; i++) {
+    allPageRequests.push($.ajax({
+      url: getControlShiftUrl(domain, category, i),
+      dataType: 'jsonp',
+      success: function(data) {
+        createCampaignMarkers(data);
+      } 
+    }))
+  }
+}
+
+function createCampaignMarkers(data) {
+  var categoryName = data.name || category;
+  var markers;
+
+  if (layerGroups[categoryName]) {
+    markers = layerGroups[categoryName];
+  } else {
+    markers = L.layerGroup();
+  }
+
+  if (data.results) {
+    data.results.forEach(function(petition) {
+      if (petition.what.length > characterLimit) {
+        petition.what = petition.what.substring(0, characterLimit) + '...';
```

Comment:
Can we use &hellip; here instead, or will this not be treated as html when it gets to the template?